File: extensions/youtube/src/main/java/app/morphe/extension/youtube/patches/FixPipChatBarPatch.java
package app.morphe.extension.youtube.patches;

import android.app.Activity;
import android.view.View;

@SuppressWarnings("unused")
public class FixPipChatBarPatch {

    /**
     * Injection point.
     */
    public static void onPipModeChanged(Activity activity, boolean isEnteringPip) {
        if (!isEnteringPip) return;
        View navBarBg = activity.findViewById(android.R.id.navigationBarBackground);
        if (navBarBg != null) navBarBg.setVisibility(View.GONE);
    }
}

File: patches/src/main/kotlin/app/morphe/patches/youtube/misc/fix/pipchatbar/FixPipChatBarPatch.kt
package app.morphe.patches.youtube.misc.fix.pipchatbar

import app.morphe.patcher.extensions.InstructionExtensions.addInstruction
import app.morphe.patcher.patch.bytecodePatch
import app.morphe.patches.youtube.misc.extension.sharedExtensionPatch

private const val EXTENSION_CLASS = "Lapp/morphe/extension/youtube/patches/FixPipChatBarPatch;"

/**
 * Hides the bar that appears over the PiP video after using live chat text entry.
 */
internal val fixPipChatBarPatch = bytecodePatch{
    dependsOn(sharedExtensionPatch)

    execute {
        PipModeChangedFingerprint.method.addInstruction(
            0,
            "invoke-static { p0, p1 }, $EXTENSION_CLASS->onPipModeChanged(Landroid/app/Activity;Z)V"
        )
    }
}
